corehq/apps/userreports/expressions/json_query_specs.py

```diff
@@ -0,0 +1,98 @@
+import itertools
+from dimagi.ext.jsonobject import JsonObject, DictProperty, StringProperty
+from corehq.apps.userreports.exceptions import BadSpecError
+from corehq.apps.userreports.specs import TypeProperty
+from jsonobject.base_properties import DefaultProperty
+from .utils import SUPPORTED_UCR_AGGREGATIONS, aggregate_items
+
+
+class FilterItemsExpressionSpec(JsonObject):
+    type = TypeProperty('filter_items')
+    items_expression = DefaultProperty(required=True)
+    filter_expression = DictProperty(required=True)
+
+    def configure(self, items_expression, filter_expression):
+        self._items_expression = items_expression
+        self._filter_expression = filter_expression
+
+    def __call__(self, doc, context=None):
+        items = self._items_expression(doc, context) or []
+
+        values = []
+        for item in items:
```

Comment:
it looks like this will fail hard if `self._items_expression` returns a non-false, non-iterable. Can you address that and add a test?